clients/venice-samza/src/main/java/com/linkedin/venice/samza/VeniceSystemProducer.java

```diff
@@ -626,6 +639,19 @@ protected CompletableFuture<Void> send(Object keyObject, Object valueObject) {
         return new Pair<>(valueSchemaResponse.getId(), valueSchemaResponse.getDerivedSchemaId());
       });
 
+      if (Version.isVersionTopic(topicName) && valueSchemaIdPair.getSecond() != -1) {
+        // This is a write compute request getting published to a version topic. We don't
+        // support partial records in the Venice version topic, so we will convert this request
+        // to a full put with default fields applied
+
+        int baseSchemaId = valueSchemaIdPair.getFirst();
+        valueObject = convertWriteComputeRecordToFullPut(
```

Comment:
What if this update record contains collection merge operation? Will we lose data?

Reply:
Interestingly, there are two scenarios to consider with collection merge.

The first is that this is only colleciton merge operation for this key.  If you do a partial update of 'any' kind during a batch load, the server will only keep the resolution of the LAST message for that given key.  This is an asymmetry that maybe we can consider fixing in the future, but right now for standardizations needs this ok and I've made it clear to them this behavior.

Otherwise this is the same logic that we run when doing a colleciton merge operation on a null record in Venice (same codepath).  The result should just be an array of all the merge elements and nothing else.